torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -634,6 +691,119 @@ llvm::Error PyTorchModelLoader::loadMax(const torch::jit::Node *ptNode) {
   return addGlowNodeValue(outputs[0], glowNode->getResult());
 }
 
+llvm::Error PyTorchModelLoader::loadSize(const torch::jit::Node *ptNode) {
+  auto inputs = ptNode->inputs();
+  auto outputs = ptNode->outputs();
+  RETURN_IF_ERR(checkInputAndOutputSizes(inputs, 2, outputs, 1));
+
+  glow::NodeValue input;
+  ASSIGN_VALUE_OR_RETURN_ERR(input,
+                             getGlowNodeValue(inputs[SizeInputs::input]));
+
+  // Index must be a Constant.
+  int32_t dim;
+  auto dimHandle = glow::Handle<int32_t>::createInvalidHandle();
+  ASSIGN_VALUE_OR_RETURN_ERR(
+      dimHandle, getGlowConstantHandle<int32_t>(inputs[SizeInputs::dim]));
+  RETURN_IF_ERR(getSingleValueFromConstant(dimHandle, dim));
+
+  if (dim == -1) {
+    dim = input.dims().size() - 1;
+  }
+
+  // Index must a valid index of input.
+  RETURN_ERR_IF_NOT(input.dims().size() - 1 >= dim,
+                    strFormat("Trying to access the size of dim %d of a tensor "
+                              "with only %d dimensions",
+                              (int32_t)dim, (int32_t)input.dims().size()));
+
+  glow::Tensor sizeT(glow::ElemKind::Int32ITy, {1});
+  sizeT.getHandle<int32_t>().raw(0) = input.dims()[dim];
+  glow::Constant *sizeConstant =
+      F_.getParent()->createConstant("size_output", std::move(sizeT));
+
+  return addGlowNodeValue(outputs[0], sizeConstant);
+}
+
+llvm::Error
+PyTorchModelLoader::loadListConstruct(const torch::jit::Node *ptNode) {
+  auto inputs = ptNode->inputs();
+  auto outputs = ptNode->outputs();
+  RETURN_IF_ERR(checkInputAndOutputSizes(inputs, -1, outputs, 1));
```

Comment:
Additional comment would be useful: -1 means no checks for inputs.

Reply:
I put this on checkInputAndOutputSizes definition, I can add a comment here too